work.py

```diff
@@ -424,8 +418,7 @@ def sync_timesheetline(cls, activities):
                 if getattr(timesheet_line, attribute, None) != value:
                     setattr(timesheet_line, attribute, value)
 
-            if (timesheet_line.work and activity.resource.timesheet_works
-                    and (timesheet_line.work != activity.resource.timesheet_works[0])):
+            if timesheet_line.work != activity.resource.timesheet_works[0]:
```

Comment:
If the "activity.resource.timesheet_works" is None, not solved current code the traceback:

```
File \"/tryton/trytond/trytond/modules/project_activity/work.py\", line 419, in sync_timesheetline\n    if timesheet_line.work != activity.resource.timesheet_works[0]:\n                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^\nIndexError: tuple index out of range\n",
```

1- Line 388,raise user warning when has duration, but user can continue:
```
activity.duration and not activity.resource.timesheet_works
```

Reply:
I saw the traceback and my patch should handle correctly that situation because it ensures that timesheet_works is not empty at the very beginning of the loop.